src/rf/apps/home/views.py

```diff
@@ -80,16 +92,20 @@ def _save_layer(request, layer, username=None):
     else:
         data = request.PUT.copy()
 
-    data['tags'] = data.getlist('tags')
-    data['images'] = data.getlist('images')
-
-    # TODO: Check if user has already created a layer with this name.
+    data['tags'] = get_list('tags', data)
+    data['images'] = get_list('images', data)
 
     form = LayerForm(data, instance=layer)
 
     if not form.is_valid():
         raise Forbidden(errors=form.errors)
 
+    if len(Layer.objects.filter(user__username=request.user.username,
+                                name=form.cleaned_data['name'])) > 0:
+        raise Forbidden(errors={
+            'duplicate_name': 'Layer with name already exists for user.'
```

Comment:
To be consistent, this format should be `'name': ['Layer with name already exists for user.']`. The `errors` format is documented here https://github.com/azavea/raster-foundry/blob/develop/src/rf/apps/core/exceptions.py#L32 but is not enforced.